mountpoint-s3/src/prefetch/parquet_prefetch.rs

```diff
@@ -8,29 +11,88 @@ use mountpoint_s3_client::ObjectClient;
 use parquet::errors::ParquetError;
 use parquet::file::footer::decode_footer;
 use parquet::file::metadata::ParquetMetaData;
+use std::ops::Range;
 use tracing::trace;
 
 use super::PrefetchReadError;
+use crate::checksums::ChecksummedBytes;
+use crate::sync::Arc;
+
+pub use async_lock::RwLock as AsyncRwLock;
 
 const PARQUET_MAGIC_LEN: usize = 8;
 
+/// Range Wrapper for sorting
+/// Ranges are considered equal if both start and end points match exactly
+/// This is used to sort ranges in the interval tree
+/// Primary comparison on start, secondary on end
+#[derive(Debug, Clone)]
+pub struct RangeKey {
+    pub range: Range<u64>,
+}
```

Comment:
This type may be a good candidate to implement [`std::ops::Deref`](https://doc.rust-lang.org/std/ops/trait.Deref.html) for. This trait is useful for smart pointers and wrappers around other types, where it makes sense for all the original behaviors of the inner type to be available.

With `RangeKey`, you're basically wrapping a `Range` so you can implement ordering on it. But it'd be great if it otherwise behaved just like the inner `Range`. That's where `Deref` comes in!

This will help you eliminate some of the references to `.range` in `parquet_stream.rs`.

If you were to implement [`From<Range>`](https://doc.rust-lang.org/std/convert/trait.From.html) for this type, then you can remove the `pub` scoping from `range` too.